src/validate_incoming.rs

```diff
@@ -53,6 +61,64 @@ impl MAuthInfo {
         }
     }
 
+    pub(crate) async fn validate_request_optionally(&self, req: Request) -> Request {
+        let (mut parts, body) = req.into_parts();
+        if parts.headers.contains_key(MAUTH_V2_SIGNATURE_HEADER)
+            || parts.headers.contains_key(MAUTH_V1_SIGNATURE_HEADER)
```

Comment:
is this check case-insensitive?

https://www.rfc-editor.org/rfc/rfc7230#section-3.2

> Each header field consists of a case-insensitive field name

Reply:
Yes, the `headers` field here is a [`http::header::HeaderMap`](https://docs.rs/http/1.2.0/http/header/index.html), not a standard HashMap. It already fully accounts for the case-insensitive nature of HTTP headers.

Reply:
nice!!